### DISCUSSION TITLE
hyperref fails to run if title contains greek letter

### DISCUSSION THREAD

**Discussion by YijunYuan** [tuna/thuthesis#942] [Q&A] [answered]
### 提交前检查

- [X] 我已经充分阅读了模板文档。
- [X] 我的问题并非关于模板的用法（否则请前往 discussions 提出）。
- [X] 我已经搜索了已有的 issues 和 discussion，确认没有重复提交。

### 系统

Ubuntu 23.10 on Windows 10 x86_64, 5.15.146.1-microsoft-standard-WSL2

### TeX 发行版

TeXLive 2023

### TeX 编译器

XeLaTeX

### 版本

7.5.0

### 学位类型

doctor

### 问题描述

以下MWE编译失败。将标题中的`\varphi`改成`x`或者去除`hyperref`宏包则可成功编译。

### 复现代码

```TeX
\documentclass{thuthesis}
\thusetup{
  title  = {$\varphi$}
}
\usepackage{hyperref}
\begin{document}
\maketitle
\frontmatter
\mainmatter
\backmatter
\end{document}
```

### 编译日志

```Text
......
Package hyperref Warning: Rerun to get /PageLabels entry.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 6.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\varphi' on input line 6.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 6.

(/usr/share/texlive/texmf-dist/tex/generic/stringenc/se-ascii-print.def)

Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 8.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 8.

! Improper alphabetic constant.
<to be read again> 
                   \mitvarphi 
l.8 \frontmatter
```

**Comment by YijunYuan** [comment]
参考 https://github.com/mohuangrui/ucasthesis/issues/154#issuecomment-531618081 ，在导言区添加
```tex
\makeatletter
\g@addto@macro\psdmapshortnames{\let\mitvarphi\textphi}
\makeatother
```
后解决。

不得不说unicode-math真的一定要加吗？

**Comment by Harry-Chen** [comment, marked as answer]
> 参考 [mohuangrui/ucasthesis#154 (comment)](https://github.com/mohuangrui/ucasthesis/issues/154#issuecomment-531618081) ，在导言区添加
> 
> ```
> \makeatletter
> \g@addto@macro\psdmapshortnames{\let\mitvarphi\textphi}
> \makeatother
> ```
> 
> 后解决。

这是比较常见的问题，请见 https://github.com/mohuangrui/ucasthesis/wiki/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98#%E7%AB%A0%E8%8A%82%E6%A0%87%E9%A2%98%E4%B8%AD%E7%9A%84%E6%95%B0%E5%AD%A6%E7%AC%A6%E5%8F%B7%E5%AF%BC%E8%87%B4%E6%8A%A5%E9%94%99。

> 不得不说unicode-math真的一定要加吗？

`unicode-math` 是现代的数学排版方案，并不仅仅用于在文中书写 unicode 码位的数学符号。

**Comment by YijunYuan** [comment]
> 
> 这是比较常见的问题，请见 [tex.stackexchange.com/questions/360607/unicode-math-but-ordinary-blackboard-bold](https://tex.stackexchange.com/questions/360607/unicode-math-but-ordinary-blackboard-bold) 。
> 

我想您引用了错误的链接。Anyway，这确实是常见问题。
